### DISCUSSION TITLE
Mirth error moving file to network path

### DISCUSSION THREAD

**Discussion by kaziff** [nextgenhealthcare/connect#5790] [Q&A] [answered]
**The channel processes the file successfully and inserts the data to sql but unable to move the file to a shared network path.**

ERROR 2023-05-19 13:49:00.133 [File Reader Polling Thread on testPhmcClinic (05aee4c2-98da-41d0-aa81-1bc49985f02a) < 05aee4c2-98da-41d0-aa81-1bc49985f02a_Worker-1] com.mirth.connect.connectors.file.FileReceiver: Error processing file in channel: 05aee4c2-98da-41d0-aa81-1bc49985f02a
com.mirth.connect.connectors.file.FileConnectorException: Error moving file from [\\ABC\DEF\XYZ\mirth\VdiTestEtl\Phmc\Clinic\Inbound/00120_CLINICS_20230424145549.txt] to [//ABC/DEF/XYZ/mirth/VdiTestEtl/Phmc/Clinic/Processed/00120_CLINICS_20230424145549.txt]
	at com.mirth.connect.connectors.file.FileReceiver.processFile(FileReceiver.java:482) ~[file-server.jar:?]
	at com.mirth.connect.connectors.file.FileReceiver.processFiles(FileReceiver.java:328) ~[file-server.jar:?]
	at com.mirth.connect.connectors.file.FileReceiver.poll(FileReceiver.java:239) ~[file-server.jar:?]
	at com.mirth.connect.donkey.server.channel.PollConnectorJob.execute(PollConnectorJob.java:49) ~[donkey-server.jar:?]
	at org.quartz.core.JobRunShell.run(JobRunShell.java:213) ~[quartz-all-2.1.7.jar:?]
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:557) ~[quartz-all-2.1.7.jar:?]

**Comment by kirbykn2** [comment]
Please provide a screenshot of the channel configuration.

On Fri, May 19, 2023 at 2:02 PM kaziff ***@***.***> wrote:

> The channel processes the file successfully and inserts the data to sql
> but unable to move the file to a shared network path.
>
> ERROR 2023-05-19 13:49:00.133 [File Reader Polling Thread on
> testPhmcClinic (05aee4c2-98da-41d0-aa81-1bc49985f02a) <
> 05aee4c2-98da-41d0-aa81-1bc49985f02a_Worker-1]
> com.mirth.connect.connectors.file.FileReceiver: Error processing file in
> channel: 05aee4c2-98da-41d0-aa81-1bc49985f02a
> com.mirth.connect.connectors.file.FileConnectorException: Error moving
> file from
> [\ABC\DEF\XYZ\mirth\VdiTestEtl\Phmc\Clinic\Inbound/00120_CLINICS_20230424145549.txt]
> to
> [//ABC/DEF/XYZ/mirth/VdiTestEtl/Phmc/Clinic/Processed/00120_CLINICS_20230424145549.txt]
> at
> com.mirth.connect.connectors.file.FileReceiver.processFile(FileReceiver.java:482)
> ~[file-server.jar:?]
> at
> com.mirth.connect.connectors.file.FileReceiver.processFiles(FileReceiver.java:328)
> ~[file-server.jar:?]
> at
> com.mirth.connect.connectors.file.FileReceiver.poll(FileReceiver.java:239)
> ~[file-server.jar:?]
> at
> com.mirth.connect.donkey.server.channel.PollConnectorJob.execute(PollConnectorJob.java:49)
> ~[donkey-server.jar:?]
> at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
> ~[quartz-all-2.1.7.jar:?]
> at
> org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:557)
> ~[quartz-all-2.1.7.jar:?]
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextgenhealthcare/connect/discussions/5790>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APRXWDYT3EAZZ5NXEXTBLNTXG6YT5ANCNFSM6AAAAAAYICODEM>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Best,

Kirby Knight
| 231.735.4650 | ***@***.***

**Reply by kaziff** [reply]
![image](https://github.com/nextgenhealthcare/connect/assets/133980287/df850c51-8f7a-4959-a274-3ec9da5e630f)

**Reply by pacmano1** [reply]
Need to show the full thing - for example for SFTP readers you can't move the file _locally_

**Reply by kaziff** [reply]
on windows, I am able to move locally. but i am unable to move from a network path

**Reply by pacmano1** [reply]
The Mirth server process owner (i.e. the user), dess it have permission to wherever you need to move the file?   Remember that is a share permission and also file permissions.

**Reply by kaziff** [reply]
Which process exactly needs permission. We gave permission to Mirth user and service account of mirth.

**Reply by kaziff** [reply]
service owner has the full control in this case

**Reply by ab-20-m** [reply]
Assuming this is a Windows SMB Share you are connecting to, the path should be using \ , not / for the component separator. Local paths in Windows can accept both, sometimes, but Microsoft says UNC is ```\\server\share\some\folders\on\share```.

**Reply by kaziff** [reply]
We checked that also. We configured source and destination with \ as in UNC, but mirth automatically changes it to /

**Reply by pacmano1** [reply]
Oops.. thought it was an ftp reader for a second.. so if you login to the server as the user running the process mirth, can you do this same file move?

**Reply by kaziff** [reply]
yes i can do this manually with same user

**Reply by pacmano1** [reply]
There is not something else removing the source file, correct?  e.g. in the channel itself.

**Reply by kaziff** [reply, marked as answer]
finally i got it. Changed local account to specified windows user account in services of mirth connect->properties.

**Reply by pacmano1** [reply]
I asked you that earlier.  "The Mirth server process owner (i.e. the user), dess it have permission to wherever you need to move the file? Remember that is a share permission and also file permissions."

**Reply by kaziff** [reply]
Yes You asked. But issue was not of permission. Windows user was ABC, and mirth service was using local system account. We had to change the owner only. Permissions were as it is. We didnt change any permission. Glad its done. Thank you for your question. In fact my infrastructure team got the hint from your questions only.

**Reply by pacmano1** [reply]
**Well yes it is an issue of permissions..**  "and mirth service was using local system account." that (Local System) is "the Mirth server process owner" 

That is a permission problem because the mirth service user is the one attempting to access the network share.

**Reply by pacmano1** [reply]
Anyway, glad you sorted it out, and I am only adding clarity for the next reader of these posts.
